index.js

```diff
@@ -168,7 +177,35 @@ class KeyringController extends EventEmitter {
    * @returns {Promise<Object>} A Promise that resolves to the state.
    */
   async submitPassword(password) {
+    // If a user is trying to submit a password, we should lock
+    // in the event that the password is incorrect
+    this.setLocked();
+
+    await this.verifyPassword(password);
     this.keyrings = await this.unlockKeyrings(password);
+
+    this.password = password;
```

Comment:
While the passphrase is already set as a private variable (so shouldn't be any new exposure introduced by this PR), this sets of bells for me.

Could we get away without keeping it in memory (since the decrypted key seems to be as well)? If not, could we at least encapsulate it further? (cf. https://github.com/MetaMask/eslint-config/pull/244)

It seems reasonable that even if so, that can be addressed in a separate PR from this.